### PR TITLE
[Convert method references Round 2]LPS-164068 

### PR DESCRIPTION
**Pull request by JanisZhang**
Hi @dantewang @tinatian ,

This PR is to use serviceTrackerMap instead of the method reference in `layout-impl`. Thanks!

https://issues.liferay.com/browse/LPS-164068

### CODE REVIEW

**Comment by catarineaguiar** [comment]
no regressions.

**Review by dantewang — APPROVED** [review]

File: modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/list/retriever/ListObjectReferenceTrackerImpl.java [hunk 1]
```diff
@@ -35,29 +35,30 @@
...
-		_listObjectReferenceFactories.remove(
-			GenericUtil.getGenericClassName(listObjectReferenceFactory));
+	@Deactivate
+	protected void deactivate() {
+		_listObjectReferenceFactoryServiceTrackerMap.close();
 	}
 
-	private final Map<String, ListObjectReferenceFactory<?>>
-		_listObjectReferenceFactories = new ConcurrentHashMap<>();
+	private volatile ServiceTrackerMap<String, ListObjectReferenceFactory<?>>
```
Comment (tinatian) [inline modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/list/retriever/ListObjectReferenceTrackerImpl.java:61]: @JanisZhang @dantewang It does not need to be volatile.

**Comment by JanisZhang** [comment]
HI @tinatian, force-pushed. Thanks!

**Comment by tinatian** [comment]
@shuyangzhou This pull is good to go.

Tina

**Comment by shuyangzhou** [comment]
See https://github.com/brianchandotcom/liferay-portal/pull/123993

**Pull request closed without merging**